dss/storage/validator.py

```diff
@@ -1,23 +1,91 @@
 """Utilities in this file are used to removed extra fields from index data before adding to the index."""
+import io
 import re
+
+import boto3
+from cloud_blobstore import BlobNotFoundError
 from jsonschema import validators
 from jsonschema import _utils, _validators
 
 import json
-from urllib.request import urlopen
-from urllib.error import URLError
-import ssl
-from typing import List
+from typing import List, Optional
 import logging
 
+from requests import RequestException
+
+from dss import Replica, Config, DSSException
+from hashlib import sha1
+import requests
+
+
+DSS_Draft4Validator = validators.create(meta_schema=_utils.load_schema("draft4"),
+                                        validators={u"$ref": _validators.ref,
+                                                    u"additionalProperties": _validators.additionalProperties,
+                                                    u"properties": _validators.properties_draft4,
+                                                    u"required": _validators.required_draft4
+                                                    },
+                                        version="draft4"
+                                        )
+
+
+class SizeLimitError(requests.RequestException):
+    "The max size content size has been exceeded"
+    pass
+
+class S3UrlCache:
+    """Caches content of arbitrary URLs the first time they are requested. Currently only supports content lengths of up
+     to a few megabytes."""
+
+    def __init__(self, logger: logging.Logger):
+        self.s3_client = boto3.client('s3')
+
+        # A URL's contents are only stored in s3 to keep the data closer to the aws lambas which use them.
+        self.blobstore, _, self.bucket = Config.get_cloud_specific_handles(Replica.aws)
+        self.logger = logger
 
-def read_json_url(link):
-    gcontext = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
-    with urlopen(link, context=gcontext) as f:
-        return json.load(f)
+    def request(self, url: str, max_size: Optional[int]=64000) -> str:
```

Comment:
Also, 64000 is much too small. Let's use 1 MiB. As per ancient convention, use powers of two, e.g. 64 * 1024.

https://en.wikipedia.org/wiki/Mebibyte

Reply:
using 1Mb